common/src/main/java/com/wynntils/services/mapdata/attributes/type/MapAttributes.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.mapdata.attributes.type;
+
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.render.type.TextShadow;
+
+public interface MapAttributes {
+    // If this is the empty string (""), then no label will be displayed
+    // null means inherit
+    String getLabel();
+
+    // If this is MapFeatureIcon.NO_ICON_ID ("none"), then no icon will be displayed
+    // null means inherit
+    String getIconId();
+
+    // 1-1000, 1000 is highest priority (drawn on top of everything else)
+    // 0 means no value specified; inherit
+    int getPriority();
+
+    // 0 means no information is available, or suitable for all combat levels
+    int getLevel();
+
+    MapVisibility getLabelVisibility();
+
+    CustomColor getLabelColor();
+
+    TextShadow getLabelShadow();
+
```

Comment:
Yeah... I went back and forth with having separate Label and Icon records, and then inlining them in the general attributes, extracting them again, and so on. 

My arguments for having them "inlined" is: 

1) We only get a single attribute class, which simplifies the overriding method. Otherwise you'd have to handle the case of someone wanting to only override label shadow but not color etc. The complexity in handling this (and making sure you do not think incorrectly in the code at any one step) rises sharply.

2) Icon and label were somewhat similar, but still in the end different enough that it did not make sense to unify them. But that would mean we should have an icon class and a label class with some redundancy nevertheless, and that would not feel satisfactory either. And if we tried to unify them, we'd get an even increased complexity with some AbstractMapAttributeAspect having basically color and visibility.

Reply:
Okay, good reasoning, let's keep it as is.